### DISCUSSION TITLE
How to return from xlwings create_report 'xlwings Book' object to be downloaded?

### DISCUSSION THREAD

**Discussion by miriam-z** [fastapi/fastapi#8929] [Questions] [answered]
#### OS (e.g. Windows 10 or macOS Sierra)
Big Sur 11.4

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
Python 3.9.5
Excel 16.50

#### Describe your issue (incl. Traceback!)
How to return part of API from create_report?

#### Include a minimal code sample to reproduce the issue (and attach a sample workbook if required!)

```Python
from fastapi import FastAPI
from xlwings.pro.reports import create_report

api = FastAPI()

@api.get("/weekly-report/{start_date}/{end_date}", response_class=FileResponse)
async def weekly(start_date, end_date):

    wb = create_report(template_path, report_path, app=app, **data) // returns an xlwings Book object

    return wb

```

**Comment by STeveShary** [comment]
Are you asking how to return not the entire `wb` object but rather an object that only has some fields?

**Comment by YuriiMotov** [comment, marked as answer]
You can either:
1. save it into file and return the path to that file `return some_file_path` ([`FileResponse` docs](https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse))
2. stream this file from the buffer in memory using [StreamingResponse](https://fastapi.tiangolo.com/advanced/custom-response/#streamingresponse)
